### PR TITLE
chore: create workspace

### PR DESCRIPTION
**Pull request by davidbarsky**

### CODE REVIEW

File: aws-sigv4/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+description = "An AWS SigV4 request signer."
+repository = "https://github.com/davidbarsky/sigv4"
+homepage = "https://github.com/davidbarsky/sigv4"
+documentation = "https://docs.rs/aws-sigv4"
+
+[dependencies]
+http = "0.2"
+http-body = "0.4"
+ring = "0.16.11"
+serde = { version = "1", features = ["derive"] }
```
Comment (rcoh) [inline aws-sigv4/Cargo.toml:19]: Is serde needed in the core?

File: aws-sigv4/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+serde_urlencoded = "0.5.1"
+bytes = "1.0"
+hex = "0.4.0"
+chrono = { version = "0.4", default-features = false, features = ["clock", "std"] }
+httparse = "1"
+tower = "0.3"
+
+[dev-dependencies]
+pretty_assertions = "0.6.1"
+serde = { version = "1.0.101", features = ["derive"] }
```
Comment (rcoh) [inline aws-sigv4/Cargo.toml]: Serde in here twice

File: aws-sigv4/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+http = "0.2"
+http-body = "0.4"
+ring = "0.16.11"
+serde = { version = "1", features = ["derive"] }
+serde_urlencoded = "0.5.1"
+bytes = "1.0"
+hex = "0.4.0"
+chrono = { version = "0.4", default-features = false, features = ["clock", "std"] }
+httparse = "1"
+tower = "0.3"
```
Comment (rcoh) [inline aws-sigv4/Cargo.toml]: Can drop this I assume?

**Review by rcoh — APPROVED** [review]
thanks!! looks like a few deps can be removed. I might also consider removing the `clock` feature from chrono and taking the signing timestamp as a parameter, at least in the core libs.

File: aws-sigv4/Cargo.toml (hunk 1 above)
Comment (davidbarsky) [reply aws-sigv4/Cargo.toml:19]: Strictly speaking, no. However, I use serde in two places here:

1. Credentials.
2. Serializing a querystring for certain kinds of payloads. That _can_ be done by hand, but I'd rather not roll that by hand now.

**Review by davidbarsky — COMMENTED** [review]
> I might also consider removing the clock feature from chrono and taking the signing timestamp as a parameter, at least in the core libs.

Agreed to some extent. I think the biggest win would be getting rid of the `Utc::now` call in the `sign` function.

File: aws-sigv4/Cargo.toml (hunk 2 above)
Comment (davidbarsky) [reply aws-sigv4/Cargo.toml]: Whoops, removed. Thanks!

File: aws-sigv4/Cargo.toml (hunk 1 above)
Comment (davidbarsky) [reply aws-sigv4/Cargo.toml:19]: For now, yeah. At 1.0? Maybe not. I'm currently using serde for [`serde_urlencoded`](https://docs.rs/serde_urlencoded/), which I _did not_ want to implement for a side project.

File: aws-sigv4/Cargo.toml (hunk 3 above)
Comment (davidbarsky) [reply aws-sigv4/Cargo.toml]: Yup! Missed that one.

**Pull request merged**